lib/screen/interactive_postcard/postcard_detail_bloc.dart

```diff
@@ -198,4 +218,35 @@ class PostcardDetailBloc
     }
     return Pair(imagePath, metadataPath);
   }
+
+  Future<bool> _showMerchProduct(AssetToken? asset, bool isViewOnly) async {
+    if (asset == null) {
+      return false;
+    }
+    final isShowConfig =
+        _remoteConfig.getBool(ConfigGroup.merchandise, ConfigKey.enable) &&
+            (_remoteConfig.getBool(
+                    ConfigGroup.merchandise, ConfigKey.allowViewOnly) ||
+                !isViewOnly);
+    if (!isShowConfig) {
+      return false;
+    }
+    try {
+      final enableMerch =
+          await _postcardService.isMerchandiseEnable(asset.tokenId ?? '');
+      return enableMerch;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  bool _enableMerch(AssetToken? asset) {
```

Comment:
please move `_enableMerch ` and `_showMerchProduct ` to Postcard Extension

Reply:
it's ok for _enableMerch but  _showMerchProduct need viewOnly which also is an async function to get, using in bloc will save a async call